### PR TITLE
Update to Trino 357

### PR DESCRIPTION
**Pull request by bogdanstate**
I am trying out TileDB for a project that uses Trino. I found this connector, which looked like exactly what we needed, except that we are using Trino 356 for our project, whereas development for this connector seems to have stopped around version 340, prior to the Trino split. I figured it was easiest to just fix the bugs resulting from the split myself.

This seems to have worked. As in, `maven compile` seems to run without issue. Most of the changes are cosmetic, except for my removal of uses of `Marker` and `Bound` when connected to trino `Range`. [This is how](https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/predicate/Range.java) `Range` is defined currently. If you look through the history you will notice both `Bound` and `Marker` were [recently deprecated](https://github.com/trinodb/trino/commit/c21a1bcde7eddb3754218e10614bcedf854bac3f). I fixed things to the best of my ability, but I make no claims to understanding what's going on there in more detail.

I also added a `violations.xml` file, shamelessly cargo-culted from [here](https://raw.githubusercontent.com/prestodb/presto/master/src/modernizer/violations.xml).

Before adding this violations file, while `mvn compile` ran just fine, `mvn package -D skipTests` still failed with:
```
[ERROR] Failed to execute goal org.gaul:modernizer-maven-plugin:2.2.0:modernizer (modernizer) on project trino-tiledb: Error opening violation file: /home/bogdan/TileDB-Trino/.mvn/modernizer/violations.xml: /home/bogdan/TileDB-Trino/.mvn/modernizer/violations.xml (No such file or directory) -> [Help 1]
```
Adding this file seems to have satisfied the compiler. 

Note that a number of tests were broken previously.  I fixed all the tests, and got to a clean build via `./mvnw package`.

### CODE REVIEW

**Comment by DimitrisStaratzis** [comment]
@bogdanstate Thank you for the PR, very nice work! 

Pease go over my comments and we would be happy to merge.

File: pom.xml [hunk 1]
```diff
@@ -4,13 +4,13 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <parent>
-        <artifactId>presto-root</artifactId>
-        <groupId>io.prestosql</groupId>
-        <version>340</version>
+        <artifactId>trino-root</artifactId>
+        <groupId>io.trino</groupId>
+        <version>356</version>
```
Comment (DimitrisStaratzis) [inline pom.xml:9]: The latest version is 357 and works fine after some testing.

File: src/main/java/io/prestosql/plugin/tiledb/TileDBSplitManager.java [hunk 2]
```diff
@@ -280,11 +282,16 @@ else if (leftOvers > 0) {
                     LOG.warn("Low > high while setting ranges."); //TODO investigate why
                     return ranges;
                 }
-                if (low != high || lowBound == upperBound) {
-                    ranges.add(new Range(
-                            new Marker(range.getType(), Optional.of(nativeValueToBlock(range.getType(), low)), lowBound),
-                            new Marker(range.getType(), Optional.of(nativeValueToBlock(range.getType(), high)), upperBound)));
+                else if (low == high) {
```
Comment (DimitrisStaratzis) [inline src/main/java/io/prestosql/plugin/tiledb/TileDBSplitManager.java:285]: This condition is the if statement passes all tests. 
```
                if (low != high || highInclusive == lowerInclusive) {
                    ranges.add(range(
                            range.getType(),
                            low,
                            lowerInclusive,
                            high,
                            highInclusive));
                }
```

File: src/main/java/io/prestosql/plugin/tiledb/TileDBClient.java [hunk 3]
```diff
@@ -11,16 +11,16 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.prestosql.plugin.tiledb;
+package io.trino.plugin.tiledb;
```
Comment (DimitrisStaratzis) [inline src/main/java/io/prestosql/plugin/tiledb/TileDBClient.java:14]: Package names need to be named as well.

Comment (bogdanstate) [reply src/main/java/io/prestosql/plugin/tiledb/TileDBClient.java:14]: Renamed directories, which is what I took this comment to request. Let me know if I missed something!

File: src/main/java/io/prestosql/plugin/tiledb/TileDBSplitManager.java (hunk 2 above)
Comment (bogdanstate) [reply src/main/java/io/prestosql/plugin/tiledb/TileDBSplitManager.java:285]: Confirmed, this passed. Changed accordingly.

File: pom.xml (hunk 1 above)
Comment (bogdanstate) [reply pom.xml:9]: Changed both trino-root and trino-testing to 357, confirmed this works too.

**Comment by bogdanstate** [comment]
Thank you for your comments / suggestions! I just went through them and adopted all your requested changes. I should mention the tests seem flaky. On my machine (Ubuntu 20.04 LTS w/ 32 GB of RAM) I get them running successfully around 50% of the time. You might want to do an audit as to what causes the flakiness. I also got a Heisenbug with a `NullPointerException` caused by one of the tests in one case (went away when re-running it). Will document if it emerges again.

**Comment by DimitrisStaratzis** [comment]
Nice work! We are aware of  the flakiness in Ubuntu. We will investigate the issue and fix it in the near future. 

Please make sure you squash some of the commits so we have a cleaner history. 

Thank you for the contribution! As you continue to explore and use TileDB please let us know if you run into any issue or have questions. You can post issues here, in our [forum](https://forum.tiledb.com) or shoot us an email at hello@tiledb.com. Also feel free to suggest any improvements you'd like to see in TileDB-Trino or our new separate repo for the facebook version [here](https://github.com/TileDB-Inc/TileDB-Presto).

**Review by DimitrisStaratzis — APPROVED** [review]

**Review by gsvic — APPROVED** [review]

**Pull request merged**